RecoEgamma/EgammaElectronAlgos/src/GsfElectronAlgo.cc

```diff
@@ -807,83 +468,78 @@ void GsfElectronAlgo::checkSetup( const edm::EventSetup & es )
   }
  }
 
-
-void GsfElectronAlgo::copyElectrons( GsfElectronCollection & outEle )
- {
-  GsfElectronPtrCollection::const_iterator it ;
-  for
-   ( it = eventData_->electrons->begin() ;
-     it != eventData_->electrons->end() ;
-     it++ )
-   { outEle.push_back(**it) ; }
- }
+reco::GsfElectronCollection GsfElectronAlgo::movedElectrons()
+{
+  reco::GsfElectronCollection outEle ;
+  for(auto& ele : eventData_->electrons) outEle.push_back(std::move(ele)) ;
+  return outEle ;
```

Comment:
I was hoping to see `return std::move(eventData_->electrons);` or similar, but see now that `eventData_->electrons` is actually a `std::list` (does it really need to be a list?).

Reply:
> In the algo, there is `removeAmbiguousElectrons` and `removeNotPreselectedElectrons` which erases electrons from the middle of the collection. That's probably why they went for the linked list, because the erasing from the middle is cheaper.

That was my impression as well from a very quick look.

> However, it is debatable if this is really necessary, in particular because there are so many iterations over the electrons, for which a vector may be faster, right? Because it stores the elements contiguously and a linked list may not. Did I get this right? What do you think?

For iterating over vector is certainly faster. Linked list stores the element via pointers to next+previous elements, so iterating over leads to "pointer chasing" for which CPUs nowadays are notoriously bad.

Erasing elements of vectors is not terrible (when done with `std::remove_if`+`std::vector::erase`, unless done very often with heavy-to-move objects), so I'd assume `vector` to be a better fit here overall. But as usual, the real answer would be given by profiling.

Reply:
Okay thanks that's what I thought. Let me see if I can do the profiling quickly in the "background", I'm also curious. Do you personally have any quick workflows to profile the reconstruction? I never know how to do this quickly...

Reply:
I've e.g. generated RECO configuration with `runTheMatrix.py` ttbar noPU workflow (10824.0, the process doesn't really matter) and used a PU RelVal file as an input (would ZEE+PU be relevant for this case?).

Reply:
Ok thank you, I can also do it like that, probably with ZEE+PU.